cmd/collector/app/handler.go

```diff
@@ -80,6 +85,51 @@ func (aH *APIHandler) saveSpan(w http.ResponseWriter, r *http.Request) {
 		} else {
 			w.WriteHeader(http.StatusOK)
 		}
+
+	case "zipkin.thrift":
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		if err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		buffer := thrift.NewTMemoryBuffer()
+		if _, err = buffer.Write(bodyBytes); err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		transport := thrift.NewTBinaryProtocolTransport(buffer)
+		_, size, err := transport.ReadListBegin()
```

Comment:
why do we need to do this manually? Can't we just use the existing thrift struct with a list of span and ask it to deserialize itself from bytes? This is going to be quite hard to maintain 100% coverage if you have to handle all these edge cases